### ISSUE TITLE
Segfault if CUDA fails to get human-readable error

### ISSUE THREAD

**Issue by Arc676** [openucx/ucx#9099]
### Describe the bug
The `UCT_CUDADRV_FUNC` macro in `src/uct/cuda/base/cuda_iface.h` assumes that `cuGetErrorString` succeeds, but according to the [documentation](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__ERROR.html#group__CUDA__ERROR_1g72758fcaf05b5c7fac5c25ead9445ada), this conversion can fail, in which case the string gets set to the null pointer instead. In the macro, UCX then tries to print this string, causing a segmentation fault by attempting to read a non-existent string.

This happens consistently on program exit during clean-up. After modifying the macro to check if the error string was obtained successfully, UCX reports the following:

```
cuda_copy_iface.c:460  UCX  ERROR cuCtxGetCurrent(&cuda_context)() failed with no error string: 4
 cuda_ipc_iface.c:472  UCX  ERROR cuCtxGetCurrent(&cuda_context)() failed with no error string: 4
```

In the case that prompted this issue, the error turned out to be CUDA error 4, which is `CUDA_ERROR_DEINITIALIZED` according to the CUDA header file.

For context, the modification was just a simple check
```cpp
if (cuGetErrorString(_result, &cu_err_str) == CUDA_SUCCESS)    \
    ucs_log((_log_level), "%s() failed: %d (%s)",              \
            UCS_PP_MAKE_STRING(_func), _result, cu_err_str);   \
else                                                           \
    ucs_log((_log_level),                                      \
            "%s() failed with no error string: %d",            \
            UCS_PP_MAKE_STRING(_func), _result);               \
```

### Steps to Reproduce
Build script:
```sh
if [ ! -d ucx ]; then
    git clone https://github.com/openucx/ucx.git
fi
cd ucx
git fetch
git checkout v1.14.1
./autogen.sh
./configure --with-pic --enable-optimizations --without-cm --without-rocm --without-xpmem --without-ugni \
    --without-java --enable-cma --enable-mt --with-verbs --with-cuda=$CUDA_HOME \
    --with-knem --with-rdmacm \
    --prefix=[...]
```

### Setup and versions
- OS version + CPU architecture: Red Hat 7.9, x86_64

- For GPU related issues:
  - GPU type: GTX 1080
  - Cuda: 
      - Drivers version: 530.30.02
      - Toolkit: 12.1.1

### Additional information (depending on the issue)
- OpenMPI version: 4.1.5 (built from source)
- Prior builds of the same project used CUDA 11.2.2 + OpenMPI 4.0.5 + UCX 1.10.0 and never encountered this issue.

**Comment by tvegas1** [comment]
Could you kindly check if #9161 fixes issue?

**Comment by Arc676** [comment]
Seems to work fine.

**Comment by tvegas1** [comment]
Merged #9161
